pkgs/development/python-modules/visions/default.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pythonOlder
+, pytestCheckHook
+, attrs
+, imagehash
+, matplotlib
+, multimethod
+, networkx
+, numpy
+, pandas
+, pillow
+, pydot
+, pygraphviz
+, shapely
+,
+}:
+buildPythonPackage rec {
+  pname = "visions";
+  version = "0.7.6";
+  disabled = pythonOlder "3.8";
+
+  format = "setuptools";
+
+  src = fetchFromGitHub {
+    owner = "dylan-profiler";
+    repo = "visions";
+    rev =
+      "5fe9dd0c2a5ada0162a005c880bac5296686a5aa"; # no tagged release for 0.7.6
+    hash = "sha256-SZzDXm+faAvrfSOT0fwwAf9IH7upNybwKxbjw1CrHj8=";
+  };
+
+  propagatedBuildInputs = [ attrs multimethod networkx numpy pandas ];
+
+  nativeCheckInputs = [ pytestCheckHook imagehash ]
+    ++ passthru.optional-dependencies.type-geometry
```

Comment:
If you would like these passthrus to be overridable, do something like

```
buildPythonPackage (final: rec {
  # snip
  ++ final.passthru.optional-dependencies.type-geometry
  ++ final.passthru...
```

Reply:
Thanks, updated!